### ISSUE TITLE
Unable to compile solution that builds fine in VS2017 with Roslyn

### ISSUE THREAD

**Issue by madelson** [dotnet/roslyn#29170]
**Version Used**: 2.8.2

**Steps to Reproduce**:
See https://stackoverflow.com/questions/51755892/errors-with-core-types-when-compiling-a-solution-using-roslyn

Key code snippet:
```
var solution = await MSBuildWorkspace.Create().OpenSolutionAsync(@"C:\path\to\solution.sln");
var project = solution.Products.Single(p => p.Name == "ProjectName");
var compilation = await project.GetCompilationAsync();
var diagnostics = compilation.GetDiagnostics()
    .Where(d => !d.IsSuppressed && d.Severity == DiagnosticSeverity.Error)
    .ToArray();
foreach (var diagnostic in diagnostics) { /* print */ }
```

**Expected Behavior**: No diagnostics (build success)

**Actual Behavior**: Hundreds of diagnostics around core types, such as `error CS0234: The type or namespace name 'SqlClient' does not exist in the namespace 'System.Data'`

**Comment by madelson** [comment]
@DustinCampbell is this known to be fixed in the lastest version or are you just suggesting I re-try it there?

**Comment by CyrusNajmabadi** [comment]
Is this still an issue with latest roslyn?
